qctrlopencontrols/cirq/cirq_circuit.py

```diff
@@ -0,0 +1,505 @@
+            raise ArgumentsValueError(
+                'Open Controls support a sequence with one '
+                'valid pulse at any offset. Found sequence '
+                'with multiple rotation operations at an offset.',
+                {'dynamic_decoupling_sequence': str(dynamic_decoupling_sequence),
+                 'instance_operation': instance_operation})
+        gate_list = []
+        for qubit in target_qubits:
+            if nonzero_pulse_counts == 0:
+                gate_list.append(cirq.I(qubit))
+            else:
+                if not np.isclose(rotations[0], 0.0):
+                    gate_list.append(cirq.Rx(rotations[0])(qubit))
+                elif not np.isclose(rotations[1], 0.0):
+                    gate_list.append(cirq.Ry(rotations[1])(qubit))
+                elif not np.isclose(rotations[2], 0.):
+                    gate_list.append(cirq.Rz(rotations[2])(qubit))
+        offset_count += 1
+        circuit.append(gate_list)
+
+    if pre_post_gate is not None:
+        gate_list = []
+        for qubit in target_qubits:
+            gate_list.append(pre_post_gate(qubit))
+        circuit.append(gate_list)
+
+    if add_measurement:
+        gate_list = []
+        for idx, qubit in enumerate(target_qubits):
+            gate_list.append(cirq.measure(qubit, key='qubit-{}'.format(idx)))
+        circuit.append(gate_list)
+
+    return circuit
+
+
+def convert_dds_to_cirq_circuit(
+        dynamic_decoupling_sequence,
+        target_qubits=None,
+        gate_time=0.1,
+        pre_post_gate_unitary_matrix=DEFAULT_PRE_POST_ROTATION_MATRIX,
+        add_measurement=True,
+        circuit_type=STANDARD_CIRCUIT,
+        device=None):
+
+    """Converts a Dynamic Decoupling Sequence into quantum circuit
+    as defined in cirq
+
+    Parameters
+    ----------
+    dynamic_decoupling_sequence : DynamicDecouplingSequence
+        The dynamic decoupling sequence
+    target_qubits : list, optional
+        List of target qubits for the sequence operation; the qubits must be
+        cirq.Qid type; defaults to None in which case a 1-D lattice of one
+        qubit is used (indexed as 0).
+    gate_time : float, optional
+        Time (in seconds) delay introduced by a gate; defaults to 0.1
+    pre_post_gate_unitary_matrix : numpy.ndarray or None, optional
+        A 2x2 unitary matrix as pre-post gate operations. Defaults to
+        the unitary matrix corresponding to a rotation of :math:'\\pi/2' around
+        X-axis. If None, pre-post gate is omitted from the circuit.
+    add_measurement : bool, optional
+        If True, the circuit contains a measurement operation for each of the
+        target qubits. Measurement from each of the qubits is associated
+        with a string as key. The string is formatted as 'qubit-X' where
+        X is a number between 0 and len(target_qubits).
+    circuit_type : str, optional
+        One of 'scheduled circuit' or 'standard circuit'. In the case of
```

Comment:
I'm not really sure about python best practice here, but instead you could refer to constants.SCHEDULED_CIRCUIT and constants.STANDARD_CIRCUIT. That might make it a bit harder to pass an invalid value (even though we don't have a compiler, people might use IDEs that warn them about using non-existent constants).

Reply:
I went with legacy here. If this needs to be changes, we have to check carefully other similar instances.